### DISCUSSION TITLE
Frigate+ available in YoloNAS-M or YoloNAS-L

### DISCUSSION THREAD

**Discussion by philippedegeer** [blakeblackshear/frigate#14977] [Ask A Question]
Goodmorning,
I am using a YoloNAS version of the Frigate+ model. Detection is great, even under difficult circumstances (low contrast, blur...). I still have some False Positives though, which I find hard to eliminate.
I my experience, and with my use case, one very effective way to eliminate FP is to move to a larger model.
Is it part of the Frigate+ roadmap to offer custom trained versions in YoloNAS-M or YoloNAS-L sizes ?
Thanks - Philippe

**Comment by blakeblackshear** [comment]
Not currently. Unless you have very powerful hardware, the inference times become too high. In my experience, the larger models mainly help with small objects. This isn't that helpful in Frigate because a cropped portion of the frame is already sent for detection making small objects no longer small in most cases. I am exporting the model as 320x320 instead of 640x640 and will probably look at adding an additional 640x640 export first for users with more powerful hardware.

For now, I will be focused on improving the performance of the existing model since I think there is quite a bit of headroom for improvement there.

**Reply by philippedegeer** [reply]
OK clear, thanks for your prompt answer

My experience with larger YoloNAS is a bit different though, at least with COCO-trained models. My case includes four camera pointing to a wild/forest area, detecting person/cat/dogs. With the various YoloNAS sizes (all in 320x320), the FP occurence has been

YoloNAS-S : approx. two or three FP per day
YoloNAS-M : approx. one FP per week
YoloNAS-L : nonexistent

I understand my observations may be anecdotal and my case quite specific but I wanted to share it with you.
